tests/integration/vpp/040_acl_test.go

```diff
@@ -0,0 +1,906 @@
+//  Copyright (c) 2019 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package vpp
+
+import (
+	"github.com/ligato/cn-infra/logging/logrus"
+	acl "github.com/ligato/vpp-agent/api/models/vpp/acl"
+	. "github.com/onsi/gomega"
+	"strings"
+	"testing"
+
+	_ "github.com/ligato/vpp-agent/plugins/vpp/aclplugin"
+	aclplugin_vppcalls "github.com/ligato/vpp-agent/plugins/vpp/aclplugin/vppcalls"
+	_ "github.com/ligato/vpp-agent/plugins/vpp/ifplugin"
+	"github.com/ligato/vpp-agent/plugins/vpp/ifplugin/ifaceidx"
+	ifplugin_vppcalls "github.com/ligato/vpp-agent/plugins/vpp/ifplugin/vppcalls"
+	//	"encoding/json"
+)
+
+func newAclIpRule(src, dst string) *acl.ACL_Rule {
+	return &acl.ACL_Rule{
+		Action: acl.ACL_Rule_PERMIT,
+		IpRule: &acl.ACL_Rule_IpRule{
+			Ip: &acl.ACL_Rule_IpRule_Ip{
+				SourceNetwork:      src,
+				DestinationNetwork: dst,
+			},
+		},
+	}
+}
+
+func newAclMacIpRule(perm bool, src string, srcPrefix uint32, srcMacAddr string, srcMacMask string) *acl.ACL_Rule {
+	var aclRulePerm acl.ACL_Rule_Action
+	if perm {
+		aclRulePerm = acl.ACL_Rule_PERMIT
+	} else {
+		aclRulePerm = acl.ACL_Rule_DENY
+
+	}
+	return &acl.ACL_Rule{
+		Action: aclRulePerm,
+		MacipRule: &acl.ACL_Rule_MacIpRule{
+			SourceAddress:        src,
+			SourceAddressPrefix:  srcPrefix,
+			SourceMacAddress:     srcMacAddr,
+			SourceMacAddressMask: srcMacMask,
+		},
+	}
+}
+
+var aclNoRules []*acl.ACL_Rule
+
+var aclErr1Rules = []*acl.ACL_Rule{
```

Comment:
You plan to remove this right? The idea was to remove the global variables altogether and just call functions from tests.

Reply:
Yes, I admit that I did not understand at first.
I eliminated all variables related to ACL_rule.